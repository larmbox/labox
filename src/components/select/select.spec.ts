import { mount } from '@vue/test-utils';
import { expect, test } from 'vitest';
import { LSelect } from '~/index';

test('mount component', () => {
  expect(LSelect).toBeTruthy();

  const component = mount(LSelect, {
    props: {
      id: 'id',
    },
  });

  expect(component.html()).toMatchSnapshot();
  expect(component.html()).toContain('select');
  expect(component.html()).toContain('form-field');
});

test('register css variables', () => {
  mount(LSelect);
  expect(
    document.getElementById('lx-theme-provider-global')!.innerText
  ).toContain('--lx-select');
});

test('inherit attributes', () => {
  const component = mount(LSelect, {
    props: { notaprop: '1', variant: 'primary' },
  });
  expect(component.html()).toContain('notaprop="1"');
  expect(component.html()).not.toContain('variant="primary"');
});
