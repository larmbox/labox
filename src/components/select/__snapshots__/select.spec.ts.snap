// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"form-control form-field select select-md select-primary\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"select-inner\\">
    <div class=\\"select-element\\"><select id=\\"id\\">
        <!--v-if-->
      </select><span class=\\"icon chevron-expand select-icon\\" aria-hidden=\\"true\\"></span></div>
  </div>
  <!--v-if-->
  <!--v-if-->
</div>"
`;
